runtime/interpreter/interpreter.go

```diff
@@ -2235,7 +2234,7 @@ func (interpreter *Interpreter) invokeInterpretedFunction(
 	// Start a new activation record.
 	// Lexical scope: use the function declaration's activation record,
 	// not the current one (which would be dynamic scope)
-	interpreter.activations.Push(function.Activation)
+	interpreter.activations.Push(activations.NewActivation(function.Activation))
```

Comment:
This pattern repeats a lot. Could it be replaced with `PushNewWithParent(function.Activation)`?